src/PythonProjectBootstrapper/ProjectGenerationUtils.py

```diff
@@ -60,14 +73,17 @@ def ConditionallyRemoveUnchangedTemplateFiles(
     new_manifest_dict: dict[str, str],
     existing_manifest_dict: dict[str, str],
     output_dir: Path,
-) -> None:
-    # Removes any template files no longer being generated as long as the file was never modified by the user
+) -> list[str]:
+    # Removes any template files no longer being generated as long as the file was never modified by the user.
```

Comment:
Nit: Consider using docstrings (rather than comments) for text that describes what the function is doing. Docstrings are used by tools (for example, IntelliSense within vscode) whereas comments are ignored.